src/main/java/net/neoforged/neoforge/event/EventHooks.java

```diff
@@ -828,37 +829,65 @@ public static void onLivingConvert(LivingEntity entity, LivingEntity outcome) {
         NeoForge.EVENT_BUS.post(new LivingConversionEvent.Post(entity, outcome));
     }
 
+    @Deprecated(since = "1.21.1", forRemoval = true)
     public static EntityTeleportEvent.TeleportCommand onEntityTeleportCommand(Entity entity, double targetX, double targetY, double targetZ) {
         EntityTeleportEvent.TeleportCommand event = new EntityTeleportEvent.TeleportCommand(entity, targetX, targetY, targetZ);
         NeoForge.EVENT_BUS.post(event);
         return event;
```

Comment:
As cleanup, could you please refactor all of the methods of this family (for `EntityTeleportEvent`) to do the creation, posting, and return in one line? As I recall correctly, this three-line pattern was because `IEventBus#post` originally returned the cancellation status of the event, but our work on Bus made it so it now returns the actual event instance.

Reply:
Sure. I had the urge to do that anyway, but I've been well trained (read: been scolded often enough) not to change more code than what's needed to implement a ticket at work... ;)